reference/shaders-hlsl/comp/ssbo-array.comp

```diff
@@ -0,0 +1,25 @@
+struct SSBO
+{
+    float4 data[1];
+};
+
+RWStructuredBuffer<SSBO> ssbos : register(u0);
+
+static uint3 gl_GlobalInvocationID;
+struct SPIRV_Cross_Input
+{
+    uint3 gl_GlobalInvocationID : SV_DispatchThreadID;
+};
+
+void comp_main()
+{
+    uint ident = gl_GlobalInvocationID.x;
+    ssbos[1].data[ident] = ssbos[0].data[ident];
```

Comment:
This will work:
```
layout(std430, binding = 0) buffer SSBO0
{
    // Something**fixed** size
} data[10];

layout(std430, binding = 1) buffer SSBO1
{
    // Something**fixed** size
} data;

layout(std430, binding = 2) buffer SSBO2
{
    // Something**fixed** size
} data[];
```

This might work but will result in accessing out of bounds on an array of size 1:
```
layout(std430, binding = 3) buffer SSBO3
{
    vec2 array[];
} data;
```

This will not work:
```
layout(std430, binding = 4) buffer SSBO4
{
    vec2 array[];
} data[10];

// The contents are too big for a RWStructuredBuffer
layout(std430, binding = 5) buffer SSBO5
{
    float buf[10000000];
} data
```

That being said, after further investigation, if we want more complete support for HLSL compute, this will all probably need to change to use `RWByteAddressBuffer` instead of `RWStructuredBuffer`. `RWByteAddressBuffer` is just a raw view of bytes and has fewer limitations than `RWStructuredBuffer`. (`RWStructuredBuffer<T>` has a limited size for `T` as well. You cannot do `RWStructuredBuffer<ReallyBigStruct>`).

Using `RWStructuredBuffer` would require converting all `OpAccessChains` for UAVs to Load/Store operations with the correct byte offsets, so that may be quite a bit of work.